File: Android/app/src/main/java/app/visafe/net/doh/Transaction.java

package app.visafe.net.doh;

import app.visafe.net.dns.DnsPacket;
import java.io.Serializable;
import java.util.Calendar;

/**
 * A representation of a complete DNS transaction, whether it succeeded or failed.
 */
public class Transaction implements Serializable {

  public enum Status {
    COMPLETE,
    SEND_FAIL,
    HTTP_ERROR,
    BAD_RESPONSE,
    INTERNAL_ERROR,
    CANCELED
  }

  public Transaction(DnsPacket query, long timestamp) {
    this.name = query.getQueryName();
    this.type = query.getQueryType();
    this.queryTime = timestamp;
  }

  public final long queryTime;
  public final String name;
  public final short type;
  public long responseTime;
  public Status status;
  public byte[] response;
  public Calendar responseCalendar;
  public String serverIp;
}

File: app/src/main/java/com/vn/visafe_android/dns/net/doh/Transaction.kt
package com.vn.visafe_android.dns.net.doh

import com.vn.visafe_android.dns.net.dns.DnsPacket
import java.io.Serializable
import java.util.*

/**
 * A representation of a complete DNS transaction, whether it succeeded or failed.
 */
class Transaction(query: DnsPacket, timestamp: Long) : Serializable {
    enum class Status {
        COMPLETE, SEND_FAIL, HTTP_ERROR, BAD_RESPONSE, INTERNAL_ERROR, CANCELED
    }

    val queryTime: Long = timestamp
    val name: String = query.queryName
    val type: Short = query.queryType
    var responseTime: Long = 0
    var status: Status? = null
    var response: ByteArray? = null
    var responseCalendar: Calendar? = null
    var serverIp: String? = null
}